src/Console/Command/MigrateGenerate.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace App\Console\Command;
+
+use App\Helper\CycleOrmHelper;
+use Spiral\Migrations\Migrator;
+use Spiral\Migrations\Config\MigrationConfig;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class MigrateGenerate extends Command
+{
+    protected static $defaultName = 'migrate/generate';
+
+    /** @var Migrator */
+    private $migrator;
+    /** @var CycleOrmHelper */
+    private $cycleHelper;
+    /** @var MigrationConfig */
+    private $config;
+
+    public function __construct(
+        Migrator $migrator,
+        MigrationConfig $conf,
+        CycleOrmHelper $cycleHelper
+    ) {
+        parent::__construct();
+        $this->migrator = $migrator;
+        $this->config = $conf;
+        $this->cycleHelper = $cycleHelper;
+    }
+
+    public function configure(): void
+    {
+        $this->setDescription('Generates a migration');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->cycleHelper->generateMigration($this->migrator, $this->config);
```

Comment:
@wolfy-j suggested that [such check is needed](https://github.com/spiral/framework/blob/master/src/Command/Cycle/MigrateCommand.php#L62).

Reply:
Cycle will generate migrations based on diff with current DB schema if the schema is not consistent you will generate broken migrations. This is a small nasty issue, so such check is recommended.

Reply:
I understand and support this recommendation. Also, it would be nice not to generate empty migrations

Reply:
Hmm... Why would empty migrations be generated?

Reply:
We are checking generated migrations via Generator observer: https://github.com/spiral/framework/blob/master/src/Command/Cycle/MigrateCommand.php#L78

You are free to copy its code, it also gives you a nice description of what have changed.

Reply:
@samdark for example: if Entities not changed then running `migrate/generate` it creates migration contains void methods (`up()` and `down()` )
@wolfy-j ok, I'll see it later, and if necessary, i'll implement the comparison code

Reply:
Fixed in 1.0.5

Reply:
One question. How would it work if you write migration manually? Would that still create dummy migration classes for me

Reply:
ORM and manual migrations works using the same engine, you can combine them if you want. Basically ORM simply writes migrations for you, the rest is the same.

Reply:
https://github.com/cycle/migrations/blob/master/src/GenerateMigrations.php#L87